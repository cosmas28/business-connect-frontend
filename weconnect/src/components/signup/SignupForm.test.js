import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import ReactDOM from 'react-dom';
import SignUpForm from './SignupForm';

describe('Sign up form', () => {
    let props = null,
        mountedSignUpForm;
    const mockRegisterUser = jest.fn(),
        mockInputChange = jest.fn();
    const component = () => {
        if (!mountedSignUpForm){
            mountedSignUpForm = mount(
                <SignUpForm {...props} />
            );
        }
        return mountedSignUpForm;
    };

    beforeEach(() => {
        props = {
            handleInputChange: mockRegisterUser,
            handleSubmitForm: mockInputChange,
            outPutSuccessMessage: null,
            outPutErrorMessage: null
        };
        mountedSignUpForm = undefined;
    });

    it('renders and matches our snapshot', () => {
        expect(component()).toMatchSnapshot();
    });

    it('contains the form', () => {
        expect(component().find('input')).toHaveLength(6);
        expect(component().find('button')).toHaveLength(1);
    });

    it('should call `handleSubmitForm` when submit button is called', () => {
        component().find('form').simulate('submit');
        expect(props.handleSubmitForm).toBeCalled();
    });

    it('should call `handleInputChange` when input field is changed', () => {
        component().find('input.test').simulate('change');
        expect(props.handleSubmitForm).toBeCalled();
    });

    describe('when `outPutSuccessMessage` is passed', () => {
        beforeEach(() => {
            props.outPutSuccessMessage = 'You have successfully created account!';
        });

        it('renders a `Success Alert box`', () => {
            const alertDiv = component().find('div.alert-success');
            expect(alertDiv.length).toBe(1);
        });
    });

    describe('when `outPutSuccessMessage` is null', () => {
        beforeEach(() => {
            props.outPutSuccessMessage = null;
        });

        it('renders a `Alert box`', () => {
            const alertDiv = component().find('div.alert-success');
            expect(alertDiv.length).toBe(0);
        });
    });

    describe('when `outPutErrorMessage` is passed', () => {
        beforeEach(() => {
            props.outPutErrorMessage = 'User already registered. Sign in';
        });

        it('renders a `Error Alert box`', () => {
            const alertDiv = component().find('div.alert-danger');
            expect(alertDiv.length).toBe(1);
        });
    });
});
