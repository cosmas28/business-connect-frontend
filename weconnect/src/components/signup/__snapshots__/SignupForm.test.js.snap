// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up form renders and matches our snapshot 1`] = `
<div
  className="col-md-6 col-sm-12 col-xs-12 offset-md-2"
>
  <h2>
    Sign up to register a business
  </h2>
  <article
    className="alert alert-success"
    role="alert"
  />
  <form
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <input
          className="input-default form-text-input"
          onChange={[Function]}
          placeholder="Email Address"
          type="email"
          value=""
        />
      </div>
      <div
        className="col"
      >
        <input
          className="input-default form-text-input"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
      </div>
    </div>
    <hr />
    <div
      className="row"
    >
      <div
        className="col"
      >
        <input
          className="input-default form-text-input"
          onChange={[Function]}
          placeholder="First name"
          type="text"
          value=""
        />
      </div>
      <div
        className="col"
      >
        <input
          className="input-default form-text-input"
          onChange={[Function]}
          placeholder="Last name"
          type="text"
          value=""
        />
      </div>
    </div>
    <hr />
    <div
      className="form-input-division"
    >
      <input
        className="input-default form-text-input"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div
      className="form-input-division"
    >
      <input
        className="input-default form-text-input"
        onChange={[Function]}
        placeholder="Confirm password"
        type="password"
        value=""
      />
    </div>
    <div
      className="form-input-division"
    >
      <button
        className="btn btn-primary input-default form-btn"
        type="submit"
      >
        Sign Up
      </button>
    </div>
    <div
      className="form-input-division"
    >
      <a
        href=""
      >
        You have an account?
      </a>
    </div>
  </form>
</div>
`;
